drivers/base/bootvid/common.c

```diff
@@ -15,8 +15,63 @@ ULONG VidpScrollRegion[4] =
     SCREEN_HEIGHT - 1
 };
 
+/*
+ * Boot video driver default palette is similar to the standard 16-color
+ * CGA palette, but it has Red and Blue channels swapped, and also dark
+ * and light gray colors swapped.
+ */
+const PALETTE_ENTRY DefaultPalette[BV_MAX_COLORS] =
+{
+    {0x00, 0x00, 0x00}, /* Black */
+    {0x80, 0x00, 0x00}, /* Red */
+    {0x00, 0x80, 0x00}, /* Green */
+    {0x80, 0x80, 0x00}, /* Brown */
+    {0x00, 0x00, 0x80}, /* Blue */
+    {0x80, 0x00, 0x80}, /* Magenta */
+    {0x00, 0x80, 0x80}, /* Cyan */
+    {0x80, 0x80, 0x80}, /* Dark Gray */
+    {0xC0, 0xC0, 0xC0}, /* Light Gray */
+    {0xFF, 0x00, 0x00}, /* Light Red */
+    {0x00, 0xFF, 0x00}, /* Light Green */
+    {0xFF, 0xFF, 0x00}, /* Yellow */
+    {0x00, 0x00, 0xFF}, /* Light Blue */
+    {0xFF, 0x00, 0xFF}, /* Light Magenta */
+    {0x00, 0xFF, 0xFF}, /* Light Cyan */
+    {0xFF, 0xFF, 0xFF}, /* White */
+};
+
+static BOOLEAN ClearRow = FALSE;
+
 /* PRIVATE FUNCTIONS **********************************************************/
 
+VOID
+NTAPI
+InitializePalette(VOID)
+{
+    ULONG i, pal;
+
+    for (i = 0; i < BV_MAX_COLORS; i++)
+    {
+        pal = (DefaultPalette[i].Red << 16) | (DefaultPalette[i].Green << 8) | DefaultPalette[i].Blue;
```

Comment:
If you follow the suggestion to have the SetPaletteEntry naturally acting on COLORREF values, then this conversion is not needed. However it'll then be needed in the InitPaletteWithTable for the bitmaps.